mcproto/types/nbt.py

```diff
@@ -0,0 +1,1399 @@
+    ) -> PayloadType | Mapping[str, PayloadType] | tuple[PayloadType | Mapping[str, PayloadType], FromObjectSchema]:
+        """Convert the NBT tag to a python object.
+
+        :param include_schema: Whether to return a schema describing the types of the original tag.
+        :param include_name: Whether to include the name of the tag in the output.
+            If the tag has no name, the name will be set to "".
+
+        :return: Either :
+            - A python object representing the payload of the tag. (default)
+            - A dictionary containing the name associated with a python object representing the payload of the tag.
+            - A tuple which includes one of the above and a schema describing the types of the original tag.
+        """
+        if type(self) is EndNBT:
+            raise NotImplementedError("Cannot convert an EndNBT tag to a python object.")
+        if type(self) in (CompoundNBT, ListNBT):
+            raise TypeError(
+                f"Use the `{type(self).__name__}.to_object()` method to convert the tag to a python object."
+            )
+        result = self.payload if not include_name else {self.name: self.payload}
+        if include_schema:
+            return result, type(self)
+        return result
+
+    @override
+    def __repr__(self) -> str:
+        if self.name:
+            return f"{type(self).__name__}[{self.name!r}]({self.payload!r})"
+        return f"{type(self).__name__}({self.payload!r})"
+
+    @override
+    def __eq__(self, other: object) -> bool:
+        """Check equality between two NBT tags."""
+        if not isinstance(other, NBTag):
+            raise NotImplementedError("Cannot compare an NBTag to a non-NBTag object.")
+        if type(self) is not type(other):
+            return False
+        return self.name == other.name and self.payload == other.payload
+
+    @override
+    def to_nbt(self, name: str = "") -> NBTag:
+        """Convert the object to an NBT tag.
+
+        ..warning This is already an NBT tag, so it will modify the name of the tag and return itself.
+        """
+        self.name = name
+        return self
+
+    @property
+    @abstractmethod
+    def value(self) -> PayloadType:
+        """Get the payload of the NBT tag in a python-friendly format."""
+        ...
+
+
+# endregion
+# region NBT tags types
+
+
+class EndNBT(NBTag):
+    """Sentinel tag used to mark the end of a TAG_Compound."""
+
+    __slots__ = ()
+
+    def __init__(self):
+        """Create a new EndNBT tag."""
+        super().__init__(0, name="")
+
+    @override
+    def write_to(self, buf: Buffer, with_type: bool = True, with_name: bool = False) -> None:
+        """Write the EndNBT tag to the buffer.
+
+        :param buf: The buffer to write to.
+        :param with_type: Whether to include the type of the tag in the serialization.
+        :param with_name: Whether to include the name of the tag in the serialization.
+        """
```

Comment:
This docstring can be left empty here. `@override` decorated methods inherit their parent docstrings, so they should only contain a docstring if it needs some extra inforamtion added.

Note: in this case, the docstring here is more specific then the one in `NBTag` though, which is weird, the params should be describedn in `NBTag`'s docstring of this methods, then leave this empty.

This also applies for `read_from`, `value`, and perhaps some other methods too.